src/core/model/reducer.ts

```diff
@@ -136,13 +138,33 @@ export const combinedReducer = (state: EditorState, action: CombinedAction) => {
             return state;
           }
 
-          // add element to new parent
-          elementToMove.parent = newUiSchema;
-          (newUiSchema as Layout).elements.splice(
-            action.index,
-            0,
-            elementToMove
-          );
+          // link child and new parent
+          elementToMove.parent = newContainer;
+          if (isLinkedLayout(newContainer)) {
+            const moveRightInSameParent =
+              action.newContainer === action.uiSchemaElement.parent &&
+              (action.newContainer as LinkedLayout).elements.indexOf(
+                action.uiSchemaElement
+              ) < action.index;
+            // we need to adapt the index as we removed the element previously
+            const indexToUse = moveRightInSameParent
+              ? action.index - 1
+              : action.index;
+            (newContainer as LinkedLayout).elements.splice(
+              indexToUse,
+              0,
+              elementToMove
+            );
+          } else if (isLinkedControl(newContainer)) {
+            newContainer.options = {
```

Comment:
I meant that the options stuff is specific for the array control, you are only checking for `isLinkedControl`. I'm fine with the big action.

Reply:
Ah I see. Att the moment this can only be called on arrays however there are multiple controls which support details. In general: Whenever something is moved into a control, the only reasonable action is to place it in a detail.